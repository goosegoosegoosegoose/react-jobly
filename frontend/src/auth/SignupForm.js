import React, { useState } from "react";
import { useHistory } from "react-router-dom";

const SignupForm = ({signup}) => {
  const INITIAL_STATE = {
    username: "",
    password: "",
    firstName: "",
    lastName: "",
    email: ""
  };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const history = useHistory();

  const handleChange = evt => {
    const { name, value } = evt.target;
    setFormData(fData => ({
      ...fData,
      [name]: value
    }))
  };

  const handleSubmit = evt => {
    evt.preventDefault();
    signup({...formData});
    setFormData(INITIAL_STATE);
    history.push("/companies");
  };

  return (
    <div>
      <h2>Sign Up</h2>
      <form onSubmit={handleSubmit}>
        <label htmlFor="username">Username</label>
        <input type="text" name="username" value={formData.username} onChange={handleChange}></input>
        <label htmlFor="password">Password</label>
        <input type="text" name="password" value={formData.password} onChange={handleChange}></input>
        <label htmlFor="firstName">First name</label>
        <input type="text" name="firstName" value={formData.firstName} onChange={handleChange}></input>
        <label htmlFor="lastName">Last name</label>
        <input type="text" name="lastName" value={formData.lastName} onChange={handleChange}></input>
        <label htmlFor="email">Email</label>
        <input type="text" name="email" value={formData.email} onChange={handleChange}></input>
        <button>Submit</button>
      </form>
    </div>
  )
}

export default SignupForm;