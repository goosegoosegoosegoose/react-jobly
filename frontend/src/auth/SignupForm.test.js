import React from "react";
import { render } from "@testing-library/react";
import SignupForm from "./SignupForm"

test("snapshot", () => {
  const { asFragment } = render(<SignupForm />);
  expect(asFragment()).toMatchSnapshot();
});