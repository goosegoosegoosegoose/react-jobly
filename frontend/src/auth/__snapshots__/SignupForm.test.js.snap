// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Sign Up
    </h2>
    <form>
      <label
        for="username"
      >
        Username
      </label>
      <input
        name="username"
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        name="password"
        type="text"
        value=""
      />
      <label
        for="firstName"
      >
        First name
      </label>
      <input
        name="firstName"
        type="text"
        value=""
      />
      <label
        for="lastName"
      >
        Last name
      </label>
      <input
        name="lastName"
        type="text"
        value=""
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        name="email"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
